### PR TITLE
[BD-35] feat: add iconbefore and iconafter props to button component

### PR DESCRIPTION
**Pull request by jignaciopm**
@abutterworth 
Ticket on Jira: https://openedx.atlassian.net/browse/PAR-267
Add iconBefore and iconAfter props to Button component

Captura de pantalla de 2021-04-12 23-56-36

Example code:
```
import { Button } from '@edx/paragon';
import { Close } from '@edx/paragon/icons';

...

<Button
  variant="brand"
  iconBefore={Close}
>
   Brand
</Button>
```

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @jignaciopm! I've created [OSPR-5733](https://openedx.atlassian.net/browse/OSPR-5733) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: src/Button/index.jsx [hunk 1]
```diff
@@ -1,8 +1,54 @@
...
+  iconAfter,
+  iconBefore,
+  ...props
+}, ref) => (
+  <Button
+    {...props}
+    className={classNames(props.className)}
+    ref={ref}
+  >
+    <span className={classNames('d-flex', 'align-items-center')}>
```
Comment (abutterworth) [inline src/Button/index.jsx:20]: Please avoid using css utility classes inside paragon components and instead prefer to update the Button.scss in this directory. Utility classes add `!important` directives to styles and so should be used sparingly within Paragon itself.

**Review by abutterworth — CHANGES_REQUESTED** [review]
Nice work. I found a couple issues:

1. Please make the size of the icon change according to the button size. In the design you can see at 'md' the height is the natural 24px (1.5rem) and at 'sm' it is 20px (1.25rem). Please also infer a reasonable size for the icon at the 'lg' button size (which doesn't exist in the design spec). 
image

2. Block level button styles have been influenced by these changes. Text should remain centered 
image

**Comment by natabene** [comment]
@jignaciopm Thank you for your contribution. @abutterworth Will you review and merge when time comes?

File: src/Button/index.jsx (hunk 1 above)
Comment (jignaciopm) [reply src/Button/index.jsx]: Fixed! @abutterworth

**Comment by jignaciopm** [comment]
Fixed! @abutterworth

File: src/Icon/_variables.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+// Icons sizes
+
+$icon-inline: .9rem !default;
```
Comment (abutterworth) [inline src/Icon/_variables.scss:4]:
```suggestion
$icon-inline: .8em !default;
```
This way inline icons will size in relation to font size.

**Review by abutterworth — CHANGES_REQUESTED** [review]
Looks great! I see an error in the tests. It looks like several snapshots need updating and a Modal test is breaking because it counts spans in a button.

- To update snapshot tests run `npm run snapshot`
- The Modal test probably needs a minor change

File: src/Icon/_variables.scss (hunk 2 above)
Comment (jignaciopm) [reply src/Icon/_variables.scss]: @abutterworth fixed!

**Comment by jignaciopm** [comment]
@abutterworth fixed all!

I had to add a class specification in Modal.test.jsx to avoid conflict with the new implemented.

**Review by abutterworth — APPROVED** [review]

**Pull request merged**